yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -687,6 +755,87 @@ export class LibP2PService<T extends P2PClientType> extends WithTracer implement
     return true;
   }
 
+  /**
+   * Validate an attestation.
+   *
+   * @param attestation - The attestation to validate.
+   * @returns True if the attestation is valid, false otherwise.
+   */
+  @trackSpan('Libp2pService.validateAttestation', (_peerId, attestation) => ({
+    [Attributes.SLOT_NUMBER]: attestation.payload.header.globalVariables.slotNumber.toString(),
+  }))
+  public async validateAttestation(peerId: PeerId, attestation: BlockAttestation): Promise<boolean> {
+    const { currentSlot, nextSlot } = await this.epochCache.getProposerInCurrentOrNextSlot();
+
+    // Check that the attestation is for the current or next slot
+    const slotNumberBigInt = attestation.payload.header.globalVariables.slotNumber.toBigInt();
```

Comment:
I feel like the core validation logic for a BlockAttestation (and BlockProposal, EpochProofQuote, and ideally Tx) should all reside in their respective classes and just take as inputs the information they need.

So in this case we could call `attestation.validate(epochCache)` and it could return a `PeerErrorSeverity`.

That would make for easier reuse elsewhere, and easier testing (since it looks like none of these are tested?)

Reply:
Yeah agreed, I can write it as such